### PR TITLE
Spotinst: Support for API Load Balancer with AWS/NLB

### PR DESCRIPTION
**Pull request by liranp**
This PR complements #9011 by adding support for API Load Balancer with AWS/NLB for the Spotinst provider.

### CODE REVIEW

File: upup/pkg/fi/cloudup/spotinsttasks/elastigroup.go [hunk 1]
```diff
@@ -637,44 +650,34 @@ func (_ *Elastigroup) create(cloud awsup.AWSCloud, a, e, changes *Elastigroup) e
...
+				if len(e.TargetGroups) > 0 {
+					tgs := e.buildTargetGroups()
 
-						group.Compute.LaunchSpecification.SetLoadBalancersConfig(cfg)
+					if group.Compute.LaunchSpecification.LoadBalancersConfig == nil {
+						group.Compute.LaunchSpecification.LoadBalancersConfig = new(aws.LoadBalancersConfig)
 					}
+
+					group.Compute.LaunchSpecification.LoadBalancersConfig.LoadBalancers =
+						append(group.Compute.LaunchSpecification.LoadBalancersConfig.LoadBalancers, tgs...)
```
Comment (rifelpet) [inline upup/pkg/fi/cloudup/spotinsttasks/elastigroup.go:680]:
```suggestion
					group.Compute.LaunchSpecification.LoadBalancersConfig.TargetGroups =
						append(group.Compute.LaunchSpecification.LoadBalancersConfig.TargetGroups, tgs...)
```

File: upup/pkg/fi/cloudup/spotinsttasks/elastigroup.go [hunk 2]
```diff
@@ -1151,50 +1150,52 @@ func (_ *Elastigroup) update(cloud awsup.AWSCloud, a, e, changes *Elastigroup) e
...
+					}
+					if group.Compute.LaunchSpecification == nil {
+						group.Compute.LaunchSpecification = new(aws.LaunchSpecification)
+					}
+					if group.Compute.LaunchSpecification.LoadBalancersConfig == nil {
+						group.Compute.LaunchSpecification.LoadBalancersConfig = new(aws.LoadBalancersConfig)
 					}
+
+					group.Compute.LaunchSpecification.LoadBalancersConfig.LoadBalancers =
+						append(group.Compute.LaunchSpecification.LoadBalancersConfig.LoadBalancers, tgs...)
```
Comment (rifelpet) [inline upup/pkg/fi/cloudup/spotinsttasks/elastigroup.go:1195]:
```suggestion
					group.Compute.LaunchSpecification.LoadBalancersConfig.TargetGroups =
						append(group.Compute.LaunchSpecification.LoadBalancersConfig.TargetGroups, tgs...)
```

**Review by rifelpet — CHANGES_REQUESTED** [review]

File: upup/pkg/fi/cloudup/spotinsttasks/elastigroup.go (hunk 1 above)
Comment (liranp) [reply upup/pkg/fi/cloudup/spotinsttasks/elastigroup.go:680]: Hi @rifelpet. Thanks for reviewing this! Please note that there is no `TargetGroups` field in the [LoadBalancersConfig](https://github.com/spotinst/spotinst-sdk-go/blob/v1.91.0/service/elastigroup/providers/aws/aws.go#L643-L648) object. It only takes a list of [LoadBalancer](https://github.com/spotinst/spotinst-sdk-go/blob/v1.91.0/service/elastigroup/providers/aws/aws.go#L650-L661) objects, and each of them defines the type of load balancer in its [Type](https://github.com/spotinst/spotinst-sdk-go/blob/v1.91.0/service/elastigroup/providers/aws/aws.go#L653) field.

File: upup/pkg/fi/cloudup/spotinsttasks/elastigroup.go (hunk 2 above)
Comment (liranp) [reply upup/pkg/fi/cloudup/spotinsttasks/elastigroup.go:1195]: Same here.

**Review by rifelpet — APPROVED** [review]

**Comment by liranp** [comment]
Thank you, @rifelpet. I really appreciate it!

**Pull request merged**